lib/recurly/subscription/add_ons.rb

```diff
@@ -27,7 +27,7 @@ def << add_on
 
         exist = @add_ons.find { |a| a.add_on_code == add_on.add_on_code }
         if exist
-          exist.quantity ||= 1 and exist.quantity += 1
+          exist.quantity ||= 1 and exist.quantity += add_on.quantity || 1
```

Comment:
This line already had poor readability. Could we perhaps break it up a little since we are in here? I know this was already in there, but can we also change `and` for `&&`?